plugins/alerting/pkg/node_backend/backend.go

```diff
@@ -84,10 +84,25 @@ var (
 	_ capabilityv1.BackendServer           = (*AlertingNodeBackend)(nil)
 )
 
-var FallbackDefaultNodeSpec = &node.AlertingCapabilitySpec{
-	RuleDiscovery: &node.RuleDiscoverySpec{
-		Enabled: true,
-	},
+var (
+	fallbackDefaultNodeSpec = &node.AlertingCapabilitySpec{
+		RuleDiscovery: &node.RuleDiscoverySpec{
+			Enabled: true,
+		},
+	}
+	specMu sync.RWMutex
+)
+
+func GetFallbackSpec() *node.AlertingCapabilitySpec {
```

Comment:
The idiomatic way to do this is using a separate file called (for example) `export_test.go` that contains:
```go
package node_backend

var FallbackDefaultNodeSpec = fallbackDefaultNodeSpec
```
Now the variable will be exported only during tests, and unexported otherwise.

Reply:
I'm not seeing how this idiomatic way can resolve the datarace idiomatically, it leads to split definitions since the mutex has to be used in both the new file and original file

Reply:
Because you don't actually want to modify this unless you're running in a test. And in the test, you can just set it before/after calling the API so it won't race.

Reply:
I'll do what you did for the metrics node_config tests, but this doesn't work with the test structure